File: examples/status.mjs
// Сервисы со сбоем прямо сейчас. Node 18+: node examples/status.mjs
// Данные: shutdown.fyi, CC BY 4.0.
const res = await fetch('https://shutdown.fyi/api/status.json');
const { country, items } = await res.json();
console.log(`Индекс связности: ${country.connectivityIndex}`);
const bad = items.filter((i) => i.status === 'degraded' || i.status === 'down');
if (!bad.length) console.log('Массовых сбоев сейчас нет.');
for (const i of bad) console.log(`${i.name}: ${i.status} — ${i.url}`);

File: examples/status.py
"""Что сейчас не так: сервисы со сбоем и сервисы, закрытые из России.

    python3 examples/status.py
Данные: shutdown.fyi, CC BY 4.0.
"""
import json
import urllib.request

BASE = "https://shutdown.fyi"


def get(path):
    req = urllib.request.Request(BASE + path, headers={"User-Agent": "shutdown-fyi-example"})
    with urllib.request.urlopen(req, timeout=20) as r:
        return json.load(r)


status = get("/api/status.json")
print("Индекс связности:", status["country"]["connectivityIndex"])

bad = [i for i in status["items"] if i["status"] in ("degraded", "down")]
print("\nСбой сейчас:" if bad else "\nМассовых сбоев сейчас нет.")
for i in bad:
    print(f"  {i['name']}: {i['status']}  {i['url']}")

print("\nНе открываются из России:")
for i in status["items"]:
    if i["kind"] != "service":
        continue
    probe = get(f"/api/status/service/{i['slug']}.json").get("probe") or {}
    if probe.get("verdict") == "blocked":
        steps = {h["host"]: h["verdict"] for h in probe.get("hosts", [])}
        print(f"  {i['name']}: {steps}")
